lib/utils.js

```diff
@@ -207,8 +206,7 @@ function setup(pluginInfo, options) {
 
 function makeCacheKey(templateCompilerPath, pluginInfo, extra) {
   let templateCompilerFullPath = require.resolve(templateCompilerPath);
-  let templateCompilerCacheKey = fs.readFileSync(templateCompilerFullPath, { encoding: 'utf-8' });
-  let cacheItems = [templateCompilerCacheKey, extra].concat(pluginInfo.cacheKeys.sort());
+  let cacheItems = [templateCompilerFullPath, extra].concat(pluginInfo.cacheKeys.sort());
```

Comment:
Sadly I don't think the path is good enough. These cache keys will be used for "warm build" caching, and this `templateCompilerFullPath` is **very likely** to be stable even when updated. I think a reasonable compromise would be to hash the contents (MD5, SHA1, etc) instead of using the full content string as a cache key.

Reply:
Updated the PR to hash the file content